docs/intro_tutorial.rst

```diff
@@ -22,14 +22,84 @@ do not, please consult the
 States
 ------
 
-(Coming soon).
+Let us picture a black box that outputs a tape with a series of characters: :math:`00101,abccd,\uparrow\downarrow\uparrow,\ldots`. This is an example of a **registry** and we will be concerned with a *quantum* theory about the processing of its information.
+
+Consider then a tuple of complex numbers :math:`(z_1,z_2,\ldots, z_n)`, one for every item that we read on this tape. The set of all these tuples  forms a complex Euclidean space on which we can define **quantum states**, square matrices :math:`\rho` that contain information about the (quantum) system. These matrices are **density operators**; they are defined by two special properties:
+
+- They are formed from the product of two square matrices, :math:`\rho = Y^*Y`, where :math:`Y^*` is the *conjugate transpose* of :math:`Y^*`. This property is called **positive semidefinite**.
+- They have unit trace: :math:`\mathrm{Tr}(\rho) = 1`
+
+:code:`toqito` can quickly check if an array is a density matrix with the function :code:`toqito.matrix_props.is_density`. There are plenty of checks one can make for arrays, they are listed in the documentation of :code:`matrix_props`.
+
+(entanglemoog picture)
+
+Quantum states :math:`\rho` characterize a particular way in which the system *entangles* the possible configurations of the register. One of the main reasons for defining quantum states like this is to have a straightforward way of expressing **mixtures** of pure states.
+
+A pure state, in terms of density matrices, is a quantum state that is built from the outer product of the unit vector with itself: :math:`\rho_{\text{pure}} = u* u \quad,` for :math:`u`  a unit vector. Mixed states generalize quantum states to conditions where we have only probabilistic knowledge of the member states. Moreover, a set of positive semi
+
+In general, mixed states will be **convex combinations** of pure quantum states. For larger systems, one may want to quickly check one has a mixture, so let's use :code:`toqito` , which verifies if the states are pure and construct their mixture:
+
+.. code-block:: python
```

Comment:
It may be preferable to use the already existing random functionality that `toqito` has to illustrate how it can be used to generate random states and ensembles of states.

Reply:
I used `random.random_state_vector(L)` to create random states. Is there a function that creates a random ensemble of states directly?